File: src/main/java/TopChefRPG/model/Lesson.java
package TopChefRPG.model;


import javax.persistence.*;
import java.util.List;

@Entity
public class Lesson {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private int id;

    @Column(name = "name")
    private String name;

    @Column(name = "strengthIncrease")
    private int strengthIncrease;

    @Column(name = "dexterityIncrease")
    private int dexterityIncrease;

    @Column(name = "creativityIncrease")
    private int creativityIncrease;

    private int luckIncrease;

    private int experienceCost;


    // variables pour l'achat
    private String ingredientName;

    private int ingredientQuantity;

    @OneToMany(targetEntity = CookLesson.class, mappedBy = "lesson")
    private List<CookLesson> cookLessons;


    public Lesson() {

    }

    public Lesson(String name, Integer strengthIncrease, Integer dexterityIncrease, Integer creativityIncrease, Integer luckIncrease, Integer experienceCost, String ingredientName, int ingredientQuantity) {

        this.name = name;
        this.strengthIncrease = strengthIncrease;
        this.dexterityIncrease = dexterityIncrease;
        this.creativityIncrease = creativityIncrease;
        this.luckIncrease = luckIncrease;
        this.experienceCost = experienceCost;
        this.ingredientName = ingredientName;
        this.ingredientQuantity = ingredientQuantity;
    }

    public int getIdLesson() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getStrengthIncrease() {
        return strengthIncrease;
    }

    public int getDexterityIncrease() {
        return dexterityIncrease;
    }

    public int getCreativityIncrease() {
        return creativityIncrease;
    }

    public int getLuckIncrease() {
        return luckIncrease;
    }

    public int getExperienceCost() {
        return experienceCost;
    }

    public String getIngredientName() { return ingredientName; }

    public int getIngredientQuantity() { return ingredientQuantity; }
}

File: Angular TopChef/TopChefRPG/src/models/lesson.ts

export class Lesson{
  private _id : number =0;
  private _name : string ="";
  private _strengthIncrease :number =0;
  private _dexterityIncrease : number = 0;
  private _creativityIncrease:number =0;
  private _luckIncrease : number =0;
  private _experienceCost : number =0;
  private _ingredientName : string ="";
  private _ingredientQuantity : number =0;


  constructor() {
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get strengthIncrease(): number {
    return this._strengthIncrease;
  }

  set strengthIncrease(value: number) {
    this._strengthIncrease = value;
  }

  get dexterityIncrease(): number {
    return this._dexterityIncrease;
  }

  set dexterityIncrease(value: number) {
    this._dexterityIncrease = value;
  }

  get creativityIncrease(): number {
    return this._creativityIncrease;
  }

  set creativityIncrease(value: number) {
    this._creativityIncrease = value;
  }

  get luckIncrease(): number {
    return this._luckIncrease;
  }

  set luckIncrease(value: number) {
    this._luckIncrease = value;
  }

  get experienceCost(): number {
    return this._experienceCost;
  }

  set experienceCost(value: number) {
    this._experienceCost = value;
  }

  get ingredientName(): string {
    return this._ingredientName;
  }

  set ingredientName(value: string) {
    this._ingredientName = value;
  }

  get ingredientQuantity(): number {
    return this._ingredientQuantity;
  }

  set ingredientQuantity(value: number) {
    this._ingredientQuantity = value;
  }


}
